### DISCUSSION TITLE
The Qwen3-Omni codec for talker

### DISCUSSION THREAD

**Discussion by wcy1122** [QwenLM/Qwen3-Omni#70] [Q&A]
Hi, thanks for this great work.
I cannot find the RVQ Qwen3-Omni codec for talker (i.e. wav2code).
Will this part be released in the future? Otherwise, it's hard to fine-tune the talker part.

**Comment by weiwchu** [comment]
I do not think the Qwen team intentionally withheld the codec to discourage fine-tuning. It is more likely that legal considerations prevented them from releasing it.

While I agree that fine-tuning becomes more difficult without access to the codec, it is still possible to fine-tune the model without it.

**Reply by emesoohc2022** [reply]
Can you share more insights on how to finetune talker without the codec?

**Reply by Luffy966** [reply]
I don't think so. Because Qwen3Omni itself undergoes hard coding when talker outputs code to code2wav, making it non-differentiable. We can only build code pseudo-labels if we want to finetune talker.
